table-api/src/main/java/io/deephaven/api/agg/spec/AggSpecCountValues.java

```diff
@@ -0,0 +1,54 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.api.agg.spec;
+
+import io.deephaven.annotations.SimpleStyle;
+import org.immutables.value.Value.Immutable;
+import org.immutables.value.Value.Parameter;
+
+/**
+ * Specifies an aggregation that outputs the count of values for each group.
+ */
+@Immutable
+@SimpleStyle
+public abstract class AggSpecCountValues extends AggSpecBase {
+
+    /**
+     * Create a new AggSpecCountValues with {@code countType} of {@code NON_NULL}.
+     *
+     * @return the agg spec
+     */
+    public static AggSpecCountValues of() {
+        return of(AggCountType.NON_NULL);
+    }
+
+    /**
+     * Create a new AggSpecCountValues with the specified {@code countType}.
+     *
+     * @param countType the count type for this aggregation
+     * @return the agg spec
+     */
+    public static AggSpecCountValues of(final AggCountType countType) {
+        return ImmutableAggSpecCountValues.of(countType);
+    }
+
+    @Override
+    public final String description() {
+        return "count type" + countType();
+    }
+
+    /**
+     * Whether {@code null} input values should be included when counting the distinct input values.
+     * 
+     * @return Whether to count nulls
+     */
+    @Parameter
+    public abstract AggCountType countType();
```

Comment:
I'll post the same sort of concerns here that I have for #6270; I think we would do better to express this in terms of a Filter. What if I want to count NULL or NAN or INFINITE? Or any combination?

Reply:
Interesting idea, would prefer to see this as a new feature suggestion.

These `agg_by` ops are companions to `Numeric` / `Basic` vector ops (and will be paralleled by `update_by` `cum_count` and `rolling_count` variants) and this filter idea isn't trivial to extend to all these routines.